geopandas/plotting.py

```diff
@@ -571,6 +570,8 @@ def plot_dataframe(
         categories.sort()
         valuemap = dict((k, v) for (v, k) in enumerate(categories))
         values = np.array([valuemap[k] for k in values[~nan_idx]])
+        if legend_kwds is not None and "fmt" in legend_kwds:
+            fmt = legend_kwds.pop("fmt")
```

Comment:
`fmt` shouldn't be used for categorical data?

Reply:
Assuming you have a limited number of unique values, you can plot floats as categorical plot. And then you might want to format them.

Reply:
i think the concept was that since legends for categorical data grab unique values from the dataframe, they tend to be strings, and even when they're not, we don't use mapclassify to generate the range labels. Put differently, we avoid reformatting here since its much easier for the user to reformat the "input" data directly

Reply:
but of course, you're right. if a user doesn't want to edit the data before plotting, they might want to reformat in the legend

Reply:
I was wrong and @jorisvandenbossche is right. This code does nothing as `fmt` in this case is never used.

Reply:
maybe i'm confused. i thought @jorisvandenbossche was asking why it was done that way